lib/types/object/index.js

```diff
@@ -118,16 +118,31 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
+
+
+            if (target[rename.from] === undefined && !(rename.from instanceof RegExp)) {
                 delete target[rename.to];
             }
+            else if (rename.from instanceof RegExp) {
+                const fromKey = Object.keys(target).find((key) => {
+
+                    return rename.from.test(key);
+                });
+
+                target[rename.to] = target[fromKey];
```

Comment:
We're going to have some weird behavior here if we encounter more than one matching key. I would think we either want to error when we encounter multiple keys matching the RegExp or convert each one of them. Currently, we will only rename the last matching one that we encounter.

Ideally, something like this:

```
const objectKeys = Object.keys(target);

for (var i = 0, il = objectKeys.length; i < il; i++) {
  if (rename.from.test(objectKeys[i]) {
    target[rename.to] = target[objectKeys[i]];

   if (!rename.options.override) {
     // don't override the key if it already exists (so validation would fail I believe)
   }

    if (!rename.options.alias) {
      delete target[objectKeys[i]];
    }
  }
}
```